### PR TITLE
MultiRole Contract

### LINKED ISSUE #370: Implement a role-based permissioning contract
We should implement a contract that can manage an arbitrary number of roles for permissioning methods each identified by a human-readable name.

### PR DESCRIPTION
**Pull request by mrice32**
Adds the MultiRole contract that provides some basic permissioning features through inheritance.

Note: this also increments the solidity compiler version.

Fixes #370.

### CODE REVIEW

File: test/MultiRole.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    // Check that only the account2 and account3 are holders of the role.
+    assert.isTrue(await multiRole.holdsRole("2", account2));
+    assert.isTrue(await multiRole.holdsRole("2", account3));
+    assert.isFalse(await multiRole.holdsRole("2", account1));
+    await multiRole.revertIfNotHoldingRole("2", { from: account2 });
+    await multiRole.revertIfNotHoldingRole("2", { from: account3 });
+    assert(await didContractThrow(multiRole.revertIfNotHoldingRole("2", { from: account1 })));
+
+    // Anyone who holds the managing role should be able to add members.
```
Comment (ptare) [inline test/MultiRole.js]: ultra nit: maybe reword this comment to "Only the holder of the managing role...", because the managing role is an exclusive role.

File: contracts/MultiRole.sol [hunk 2]
```diff
@@ -0,0 +1,144 @@
+/*
+  MutliRole contract.
```
Comment (ptare) [inline contracts/MultiRole.sol]: typo: Mutli

File: test/MultiRole.js (hunk 1 above)
Comment (mrice32) [reply test/MultiRole.js]: Done.

File: contracts/MultiRole.sol (hunk 2 above)
Comment (mrice32) [reply contracts/MultiRole.sol]: Done.
